src/multiplex/client.rs

```diff
@@ -323,6 +323,30 @@ where
     }
 }
 
+/// Handles executing the service error handler in case awaiting the `ClientInner` Future panics.
+struct ClientInnerCleanup<Request, T, E, F>
+where
+    T: Sink<Request> + TryStream,
+    E: From<Error<T, Request>>,
+    F: FnOnce(E),
```

Comment:
why are all these bounds needed here? I can see why they might be needed in the Drop impl.

Reply:
Oh yeah you're right...I am not sure why but I assume there is probably a good reason.